### DISCUSSION TITLE
Set up with PM2 - webhook call script, script calls PM2 - now I have two PM2 God Daemons running!

### DISCUSSION THREAD

**Discussion by Req** [adnanh/webhook#714] [Q&A]
My GH webhook call on push works and it calls `update.sh` . The update script contains `pm2 restart all`.  This causes the following journalctl log:

```
[2024-12-04 14:32:03] Restarting PM2 services
[PM2][Initialization] Environment variable HOME (Linux) or HOMEPATH (Windows) are not set!
[PM2][Initialization] Defaulting to /etc/.pm2
```

However, in my script I do have `HOME=/root` along with `PATH=...` to make the pm2 binary work! Everything is running with the `root` user. Here's the two PM2 instances;

```
# ps aux |grep pm2
root        2959  0.0  1.9 858720 77440 ?        Ssl  Dec03   0:39 PM2 v5.4.3: God Daemon (/root/.pm2)
root       31383  0.0  1.5 781696 61320 ?        Ssl  14:32   0:00 PM2 v5.4.3: God Daemon (/etc/.pm2)
```

Have I set something up wrong here? Is there another approach?

**Comment by adnanh** [comment]
Can you paste the entire contents of your update.sh script?
